packages/ui/src/components/toggle-group.tsx

```diff
@@ -1,84 +1,160 @@
-import * as React from 'react'
+import {
+  ComponentPropsWithoutRef,
+  createContext,
+  ElementRef,
+  FC,
+  forwardRef,
+  ReactNode,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState
+} from 'react'
 
-import { toggleVariants } from '@/components'
-import { useTheme } from '@/context'
+import { Button, ButtonSizes, ButtonVariants, Tooltip, TooltipProps } from '@/components'
+import { cn } from '@/utils'
 import * as ToggleGroupPrimitive from '@radix-ui/react-toggle-group'
-import { cn } from '@utils/cn'
-import { cva, type VariantProps } from 'class-variance-authority'
-
-const toggleGroupVariants = cva('flex items-center justify-center gap-1', {
-  variants: {
-    variant: {
-      default: '',
-      outline: '',
-      compact: 'gap-0 divide-x rounded border'
-    }
-  },
-  defaultVariants: {
-    variant: 'default'
-  }
-})
 
-const ToggleGroupContext = React.createContext<VariantProps<typeof toggleVariants | typeof toggleVariants>>({
+type ToggleGroupVariant = 'outline-primary' | 'outline-secondary' | 'ghost-primary' | 'ghost-secondary'
+type ToggleTooltipProps = Pick<TooltipProps, 'title' | 'content' | 'side' | 'align'>
+
+interface ToggleGroupContextValue {
+  variant: ToggleGroupVariant
+  size: ButtonSizes
+  disabled?: boolean
+  selectedValues: string[]
+}
+
+const ToggleGroupContext = createContext<ToggleGroupContextValue>({
+  variant: 'outline-primary',
   size: 'default',
-  variant: 'default'
+  selectedValues: []
 })
 
-type ToggleGroupProps = React.ComponentPropsWithoutRef<typeof ToggleGroupPrimitive.Root> &
-  VariantProps<typeof toggleVariants | typeof toggleVariants>
-
-const ToggleGroupRoot = React.forwardRef<React.ElementRef<typeof ToggleGroupPrimitive.Root>, ToggleGroupProps>(
-  ({ className, variant, size, children, ...props }, ref) => (
-    <ToggleGroupPrimitive.Root
-      ref={ref}
-      className={cn(
-        toggleGroupVariants({
-          variant: variant
-        }),
-        className
-      )}
-      {...props}
-    >
-      <ToggleGroupContext.Provider value={{ variant, size }}>{children}</ToggleGroupContext.Provider>
-    </ToggleGroupPrimitive.Root>
-  )
-)
+export interface ToggleGroupProps
+  extends Omit<ComponentPropsWithoutRef<typeof ToggleGroupPrimitive.Root>, 'defaultValue' | 'type' | 'unselectable'> {
```

Comment:
why omit `unselectable` and `type`, only to then add them again? are the types different?